### PR TITLE
scheduledlogging: shorten TestCaptureIndexUsageStats run time

### LINKED ISSUE #87772: scheduledlogging: TestCaptureIndexUsageStats is unreasonably long to run
**Describe the problem**

This test goes through 4 iterations of waiting for 20 seconds each time for something to happen.

This is abnormally long for a unit test, and is not even necessary.

**Expected behavior**

The test can be modified to run under 10 seconds in total.

cc @maryliag @THardy98 

Jira issue: CRDB-19525

### PR DESCRIPTION
**Pull request by THardy98**
Resolves: #87772

Previously, TestCaptureIndexUsageStats ran through 4 iterations of 20 seconds for a run time of over a minute. This change reduces the run time of the test to under 10 seconds.

Release note: None

### CODE REVIEW

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (knz) [inline pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:159]: Why do you need this? Can't you have the main code signal to the test when it's finished writing to logs?

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (knz) [inline pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:184]: ditto

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (knz) [inline pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:205]: ditto

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (THardy98) [reply pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:159]: https://github.com/cockroachdb/cockroach/pull/89319/files#diff-567a23559822ff60b8d9d63b7fda0e82a78d11fa6dced531e3f6d24966282362R165

The main code signals to the channel here, after the logging has completed here: https://github.com/cockroachdb/cockroach/pull/89319/files#diff-567a23559822ff60b8d9d63b7fda0e82a78d11fa6dced531e3f6d24966282362R153.

 For some reason - that I have not been able to figure out why - the logs don't appear if you check immediately, despite logging being completed. The 1 second buffer seems to consistently allow for the logs to appear in the log file, including under stress.

Comment (knz) [reply pkg/sql/scheduledlogging/captured_index_usage_stats_test.go:159]: maybe you're missing `log.Flush`?

Comment (THardy98) [reply pkg/sql/scheduledlogging/captured_index_usage_stats_test.go]: Yup, was missing `log.Flush`, will remove the `time.Sleep` calls.

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (THardy98) [reply pkg/sql/scheduledlogging/captured_index_usage_stats_test.go]: removed `time.Sleep`

File: pkg/sql/scheduledlogging/captured_index_usage_stats_test.go
Comment (THardy98) [reply pkg/sql/scheduledlogging/captured_index_usage_stats_test.go]: removed `time.Sleep`

**Review by knz — COMMENTED** [review]
No objection from me, but please get a review from your own team and teach them what's going on here.

> Reviewed 3 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89319#-:-NDciSC36Mcveuv1VSIh:bz380t8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @THardy98)

**Review by xinhaoz — COMMENTED** [review]
What was the reasoning behind using a channel directly in the knobs over a callback function (that then does the same thing, but keeps the channel local to the test). Just wondering if a callback might offer some more flexibility in the future.

**Comment by THardy98** [comment]
> What was the reasoning behind using a channel directly in the knobs over a callback function (that then does the same thing, but keeps the channel local to the test). Just wondering if a callback might offer some more flexibility in the future.

None, I hadn't considered a callback function though I think it's a better idea for the reasons you mentioned. Changed to use a callback instead.

**Comment by xinhaoz** [comment]
*[`pkg/sql/scheduledlogging/captured_index_usage_stats.go` line 116 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/89319#-NDe1GGGBdLBFEHSLUQM:-NDe1GGGBdLBFEHSLUQN:b1bl9u7) ([raw file](https://github.com/cockroachdb/cockroach/blob/127b34f86cc9956b79517125d6490c1d1d018b39/pkg/sql/scheduledlogging/captured_index_usage_stats.go#L116)):*
> ```Go
> 	}
> 	// Otherwise, schedule the next interval normally.
> 	return telemetryCaptureIndexUsageStatsInterval.Get(&s.st.SV) - s.getLoggingDuration()
> ```

Can we save the result of `s.getLoggingDuration()` prior to this block? I guess there is an off chance that the second call here will actually result in a negative value otherwise.



_Code quote:_
```Go
	if s.getLoggingDuration() >= telemetryCaptureIndexUsageStatsInterval.Get(&s.st.SV) {
		return s.durationOnOverlap()
	}
	// Otherwise, schedule the next interval normally.
	return telemetryCaptureIndexUsageStatsInterval.Get(&s.st.SV) - s.getLoggingDuration()
```

**Review by THardy98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89319)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @xinhaoz)

___
*[`pkg/sql/scheduledlogging/captured_index_usage_stats.go` line 116 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/89319#-NDe1GGGBdLBFEHSLUQM:-NDhaklTEUjpk678wGO_:b-896fjb) ([raw file](https://github.com/cockroachdb/cockroach/blob/127b34f86cc9956b79517125d6490c1d1d018b39/pkg/sql/scheduledlogging/captured_index_usage_stats.go#L116)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Can we save the result of `s.getLoggingDuration()` prior to this block? I guess there is an off chance that the second call here will actually result in a negative value otherwise.
</blockquote></details>

Done

**Review by xinhaoz — APPROVED** [review]
LGTM, just wondering if anyone knows the answer to the question below.

File: pkg/sql/scheduledlogging/captured_index_usage_stats.go
Comment (xinhaoz) [inline pkg/sql/scheduledlogging/captured_index_usage_stats.go:150]: I'm just wondering out loud here, but I noticed this timer was created directly from the std lib over the one wrapped in timeutil pkg. I recall reading that the timeutil one had a fix relating to Reset or something. Does it matter which one we're using when the return value of Reset isn't being used? 🤔

**Comment by THardy98** [comment]
TYFR :)

**Comment by maryliag** [comment]
looks like your backports failed to be created, can you create them? (you don't need one for 22.2.0)

**Pull request merged**
